src/solver/conv_ck_igemm_fwd_bias_activ.cpp

```diff
@@ -0,0 +1,445 @@
+                                                          args.dilation,
+                                                          args.lPadding,
+                                                          args.rPadding,
+                                                          {},
+                                                          {},
+                                                          {});
+    return conv_ptrs[i]->IsSupportedArgument(argument_ptr.get());
+}
+
+template <typename DataType>
+bool ConvCKIgemmFwdBiasActiv::CheckCKApplicability(const ProblemDescription& problem) const
+{
+    std::vector<ck::tensor_operation::device::DeviceConvFwdBiasReluPtr> conv_ptrs;
+    ck::tensor_operation::device::instance::
+        add_device_conv2d_fwd_xdl_c_shuffle_bias_relu_nhwc_kyxc_nhwk_f16_instances(conv_ptrs);
+    assert(!conv_ptrs.empty());
+    const auto& args = CKArgs{problem};
+    for(const auto& it : conv_ptrs)
+    {
+        auto argument_ptr = it->MakeArgumentPointer(nullptr,
+                                                    nullptr,
+                                                    nullptr,
+                                                    nullptr,
+                                                    args.N,
+                                                    args.K,
+                                                    args.C,
+                                                    args.input,
+                                                    args.filter,
+                                                    args.output,
+                                                    args.strides,
+                                                    args.dilation,
+                                                    args.lPadding,
+                                                    args.rPadding,
+                                                    {},
+                                                    {},
+                                                    {});
+        if(it->IsSupportedArgument(argument_ptr.get()))
+            return true;
+    }
+    return false;
+}
+
+template <typename DataType>
+void RunCKSolution(const Handle& handle,
+                   const AnyInvokeParams& primitive_parameters,
+                   const ProblemDescription& problem,
+                   const PerformanceConfigConvCKIgemmFwdBiasActiv& config)
+{
+    const auto& args = CKArgs{problem};
+    std::vector<ck::tensor_operation::device::DeviceConvFwdBiasReluPtr> conv_ptrs;
+    ck::tensor_operation::device::instance::
+        add_device_conv2d_fwd_xdl_c_shuffle_bias_relu_nhwc_kyxc_nhwk_f16_instances(conv_ptrs);
+
+    // we need to add unique_id since ck's GetTypeString() does not give unique name of the kernel.
+    int unique_id = 0;
+    for(; unique_id < conv_ptrs.size(); unique_id++)
+    {
+        if(conv_ptrs[unique_id]->GetTypeString() + "_" + std::to_string(unique_id) ==
```

Comment:
This does not align with the comment above? Perhaps you meant to use `GetTypeId` ?

Reply:
I will have to remove all the hacks with `unique_id` since CK will be fixing the `GetTyepString()`